lib/ansible/playbook/base.py

```diff
@@ -440,7 +440,15 @@ def get_validated_value(self, name, attribute, value, templar):
         if attribute.isa == 'string':
             value = to_text(value)
         elif attribute.isa == 'int':
-            value = int(value)
+            msg = ''
+            if isinstance(value, string_types) and value.lstrip('-').isnumeric():
+                value = int(value)
+            elif isinstance(value, float):
+                msg = "a 'float', which we cannot cast as it would truncate"
+            else:
+                msg = f"'{value}' of type {type(value)} which is not cleanly castable into an int"
```

Comment:
This produces a confusing `The keyword 'throttle' is required to be of type 'int', but got '0' of type <class 'int'> which is not cleanly castable into an int`: *https://dev.azure.com/ansible/ansible/_build/results?buildId=121618&view=logs&j=bb77140b-c48a-59e0-3d98-c138ebdce3df&t=40ae2d96-445f-5435-416a-957d0826a183&l=559*

Reply:
yep, i missed the 'its already an int!' case